### ISSUE TITLE
Prediction for train data

### ISSUE THREAD

**Issue by santoshpalaskar77** [awslabs/gluonts#2009]
## Description
(A clear and concise description of what the feature is.)
- Package is not giving predictions for the train/past data
- Not able to analyze fitting for train data 
- If it exists, please let me know how to use it

**Comment by lostella** [comment]
Hi @santoshpalaskar77 two ways you can do this currently:
- Use [`make_evaluation_predictions`](https://github.com/awslabs/gluon-ts/blob/0dffd78ec13df7693ef6c3f56e856f76b6f91aa8/src/gluonts/evaluation/backtest.py#L36), by passing the training dataset, to get forecasts on the trailing `prediction_length` window. Then you can do anything you want with forecasts, like visualizing them or measuring accuracy.
- Use [`backtest_metrics`](https://github.com/awslabs/gluon-ts/blob/0dffd78ec13df7693ef6c3f56e856f76b6f91aa8/src/gluonts/evaluation/backtest.py#L118), by passing the training dataset, to directly get evaluation metrics on the same trailing windows. Note that this relies on `make_evaluation_predictions`, just directly does evaluations on top of it.

Please be aware that the final window of each time series in the training data has probably been used for training purposes, therefore optimizing too much for accuracy there may lead to overfitting.
